rust/src/lib/ovsdb/global_conf.rs

```diff
@@ -33,46 +36,204 @@ fn convert_map(
     ret
 }
 
-impl From<&MergedOvsDbGlobalConfig> for OvsDbUpdate {
-    fn from(ovs_conf: &MergedOvsDbGlobalConfig) -> Self {
-        let mut row = HashMap::new();
-        let mut value_array = Vec::new();
-        for (k, v) in ovs_conf.external_ids.iter() {
-            if let Some(v) = v {
-                value_array.push(Value::Array(vec![
-                    Value::String(k.to_string()),
-                    Value::String(v.to_string()),
-                ]));
+impl OvsDbConnection {
+    pub(crate) fn get_global_conf(
+        &mut self,
+    ) -> Result<OvsDbGlobalConfig, NmstateError> {
+        let reply = self.transact(&OvsDbMethodTransact {
+            db_name: OVS_DB_NAME.to_string(),
+            operations: vec![OvsDbOperation::Select(OvsDbSelect {
+                table: GLOBAL_CONFIG_TABLE.to_string(),
+                conditions: vec![],
+                columns: Some(vec!["external_ids", "other_config"]),
+            })],
+        })?;
+
+        if let Value::Array(reply) = reply {
+            if let Some(global_conf) = reply
```

Comment:
Same, this can be `if let Value::Array(Some(global_conf)) = reply`.

Reply:
Changed to
```rust
        if let Some(global_conf) = reply
            .as_array()
            .and_then(|reply| reply.first())
            .and_then(|v| v.as_object())
            .and_then(|v| v.get("rows"))
            .and_then(|v| v.as_array())
            .and_then(|v| v.first())
            .and_then(|v| v.as_object())
        {
```